### PR TITLE
Decrement users

### LINKED ISSUE #389: Deleting an investor or expert for a category does not decrement category count
When a user leaves a category, the category needs to decrement the number of users it has

### PR DESCRIPTION
**Pull request by ritmatter**
Fixes #389

### CODE REVIEW

File: api/handlers/user.js [hunk 1]
```diff
@@ -200,38 +201,49 @@ var UserHandler = {
...
-                      winston.log('info', 'Saved user: %s', userId);
-                      return res.status(200).send(user);
-                    }
-                  });
-                }
-              });
-            }
+
+          // Decrement the number of investors in the category
+          var category = Category.findByName(categoryName).then(function(category) {
```
Comment (ritmatter) [inline api/handlers/user.js]: Exact same but we find the category first and decrement

File: api/models/Category.js [hunk 2]
```diff
@@ -5,8 +5,6 @@ var Schema = mongoose.Schema;
 
 var CategorySchema = new Schema({
   name          : { type: String, required: true, unique: true },
-  repsLiquid    : { type: Number, default: 0, required: true },
```
Comment (ritmatter) [inline api/models/Category.js]: We don't need these fields

File: api/models/User.js [hunk 3]
```diff
@@ -123,7 +123,6 @@ var UserSchema = new Schema({
 
 UserSchema.pre('validate', function(next) {
   var user = this;
-  winston.log('info', 'Validating user: %s', (String) (user._id));
```
Comment (ritmatter) [inline api/models/User.js]: This message was extremely annoying

File: api/routes/TransactionRoutes.js [hunk 4]
```diff
@@ -42,136 +42,103 @@ module.exports = function(router, isAuthenticated, acl) {
       category    : categoryName
     });
```
Comment (ritmatter) [inline api/routes/TransactionRoutes.js]: Removed any references to the categoryPromise or the category

**Comment by krazemon** [comment]
LGTM

**Pull request merged**